ports/servoshell/prefs.rs

```diff
@@ -699,13 +690,18 @@ pub(crate) fn parse_command_line_arguments(args: Vec<String>) -> ArgumentParsing
 
     let mut debug_options = DiagnosticsLogging::default();
     for debug_string in cmd_args.debug {
-        let result = debug_options.extend(debug_string);
-        if let Err(error) = result {
-            println!("error: unrecognized debug option: {}", error);
+        if let Err(error) = debug_options.extend(debug_string) {
+            println!("error: {}", error);
             return ArgumentParsingResult::ErrorParsing;
         }
     }
 
+    // Check if help was requested via env var or -Z flag
+    if debug_options.help {
+        DiagnosticsLogging::print_debug_options_usage("servo");
+        return ArgumentParsingResult::Exit;
+    }
```

Comment:
I would move this exit to `extend_from_string` which can just call https://doc.rust-lang.org/std/process/fn.exit.html.